backend/tests/integration/services/test_pipeline_service.py

```diff
@@ -319,3 +319,87 @@ def test_delete_active_pipeline(
 
         assert excinfo.value.resource_type == ResourceType.PIPELINE
         assert excinfo.value.resource_id == str(pipeline_id)
+
+    def test_get_pipeline_metrics_success(self, fxt_pipeline_service, fxt_db_models, db_session):
+        """Test successfully retrieving pipeline metrics."""
+        pipeline_id = uuid4()
+        db_session.add(fxt_db_models[0])
+        db_session.flush()
+
+        model_id = fxt_db_models[0].id
+        pipeline_db = PipelineDB(
+            id=str(pipeline_id),
+            name="Test Pipeline",
+            model_id=model_id,
+        )
+        db_session.add(pipeline_db)
+        db_session.flush()
+
+        # Mock metrics collector
+        with patch(
+            "app.services.metrics_collector.MetricsCollector.get_latency_measurements"
+        ) as mock_metrics_collector:
```

Comment:
This looks like a unit test. Integration tests shouldn't mock core parts of the components being tested.